src/devices/Mpu9250/Mpu6500.cs

```diff
@@ -0,0 +1,907 @@
+        /// Get the accelerometer in G
+        /// </summary>
+        /// <remarks>
+        /// Vector axes are the following:
+        ///    +Z   +Y
+        ///  \  |  /
+        ///   \ | /
+        ///    \|/
+        ///    /|\
+        ///   / | \
+        ///  /  |  \
+        ///         +X
+        /// </remarks>
+        public Vector3 GetAccelerometer() => GetRawAccelerometer() * AccelerationScale;
+
+        private Vector3 GetRawAccelerometer()
+        {
+            Span<byte> rawData = stackalloc byte[6] { 0, 0, 0, 0, 0, 0 };
+            Vector3 ace = new Vector3();
+            ReadBytes(Register.ACCEL_XOUT_H, rawData);
+            ace.X = BinaryPrimitives.ReadInt16BigEndian(rawData);
+            ace.Y = BinaryPrimitives.ReadInt16BigEndian(rawData.Slice(2));
+            ace.Z = BinaryPrimitives.ReadInt16BigEndian(rawData.Slice(4));
+            return ace;
+        }
+
+        /// <summary>
+        /// Set or get the accelerometer low power mode
+        /// </summary>
+        public AccelerometerLowPowerFrequency AccelerometerLowPowerFrequency
+        {
+            get { return (AccelerometerLowPowerFrequency)ReadByte(Register.LP_ACCEL_ODR); }
+            set { WriteRegister(Register.LP_ACCEL_ODR, (byte)value); }
+        }
+
+        #endregion
+
+        #region Gyroscope
+
+        /// <summary>
+        /// Gyroscope bias data
+        /// </summary>
+        public Vector3 GyroscopeBias => _gyroscopeBias;
+
+        /// <summary>
+        /// Get or set the gyroscope range
+        /// </summary>
+        public GyroscopeRange GyroscopeRange
+        {
+            get => _gyroscopeRange;
+
+            set
+            {
+                WriteRegister(Register.GYRO_CONFIG, (byte)((byte)value << 3));
+                _gyroscopeRange = (GyroscopeRange)(ReadByte(Register.GYRO_CONFIG) >> 3);
+                if (_gyroscopeRange != value)
+                    throw new IOException($"Can set {nameof(GyroscopeRange)}, desired value {value}, stored value {_gyroscopeRange}");
+            }
+        }
+
+        /// <summary>
+        /// Get or set the gyroscope bandwidth
+        /// </summary>
+        public GyroscopeBandwidth GyroscopeBandwidth
+        {
+            get => _gyroscopeBandwidth;
+
+            set
+            {
+                if (value == GyroscopeBandwidth.Bandwidth8800HzFS32)
+                {
+                    WriteRegister(Register.GYRO_CONFIG, (byte)((byte)GyroscopeRange | 0x01));
+                }
+                else if (value == GyroscopeBandwidth.Bandwidth3600HzFS32)
+                {
+                    WriteRegister(Register.GYRO_CONFIG, (byte)((byte)GyroscopeRange | 0x02));
+                }
+                else
+                {
+                    WriteRegister(Register.GYRO_CONFIG, (byte)GyroscopeRange);
+                    WriteRegister(Register.CONFIG, (byte)value);
+                }
+
+                var retConf = ReadByte(Register.GYRO_CONFIG);
```

Comment:
not sure I follow this logic here, any chance you can add some pointers to the documentation about this?